File: test/org/jetbrains/android/dsl/tests/functional/HelperConstructorTest.java
package org.jetbrains.android.dsl.tests.functional;

import org.jetbrains.android.dsl.BaseGeneratorProps;
import org.jetbrains.android.dsl.Subsystem;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class HelperConstructorTest extends BaseFunctionalTest {
    private final String testDataFile = "org/jetbrains/android/dsl/tests/functional/HelperConstructorTest.kt";

    @Override
    @BeforeMethod
    public void setUp() throws Exception {
        super.setUp();
    }

    protected void initSettings(BaseGeneratorProps settings) {
        settings.setGenerateViewHelperConstructors(true);
    }

    @Test
    public void testHelperConstructors() throws Exception {
        runFunctionalTest(testDataFile, Subsystem.VIEWS);
    }
}

File: test/org/jetbrains/android/dsl/functional/HelperConstructorTest.kt
package org.jetbrains.android.dsl.functional
import org.jetbrains.android.dsl.*
import org.testng.annotations.Test
public class HelperConstructorTest : AbstractFunctionalTest() {
    private val testDataFile = "HelperConstructorTest.kt"
    override fun initSettings(settings: BaseGeneratorConfiguration) {
        settings.tunes.add(ConfigurationTune.HELPER_CONSTRUCTORS)
    }
    [Test] public fun testHelperConstructorTestFor21() {
        runFunctionalTest(testDataFile, KoanFile.VIEWS, "21")
    }
    [Test] public fun testHelperConstructorTestFor21s() {
        runFunctionalTest(testDataFile, KoanFile.VIEWS, "21s")
    }
}
